lib/src/ansi_code.dart

```diff
@@ -0,0 +1,307 @@
+// Copyright 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// TODO(kevmoo) provide library documentation.
+library ansi_code;
+
+import 'dart:async';
+import 'dart:io' as io;
+
+/// Returns `true` if formatted ANSI output is enabled for [wrapWith] and
+/// [AnsiCode.wrap].
+///
+/// By default, returns `true` if both `stdout.supportsAnsiEscapes` and
+/// `stderr.supportsAnsiEscapes` from `dart:io` are `true`.
+///
+/// The default can be overridden by setting the [Zone] variable [AnsiCode] to
+/// either `true` or `false`.
+///
+/// [overrideAnsiOutput] is provided to make this easy.
+bool get ansiOutputEnabled =>
+    Zone.current[AnsiCode] as bool ??
+    (io.stdout.supportsAnsiEscapes && io.stderr.supportsAnsiEscapes);
+
+/// Allows overriding [ansiOutputEnabled] to [enableAnsiOutput] for the code run
+/// within [body].
+T overrideAnsiOutput<T>(bool enableAnsiOutput, T body()) =>
+    runZoned(body, zoneValues: <Object, Object>{AnsiCode: enableAnsiOutput});
+
+/// The type of code represented by [AnsiCode].
+class AnsiCodeType {
+  final String _name;
+
+  /// A foreground color.
+  static const AnsiCodeType foreground = const AnsiCodeType._('foreground');
+
+  /// A style.
+  static const AnsiCodeType style = const AnsiCodeType._('style');
+
+  /// A background color.
+  static const AnsiCodeType background = const AnsiCodeType._('background');
+
+  /// A reset value.
+  static const AnsiCodeType reset = const AnsiCodeType._('reset');
+
+  const AnsiCodeType._(this._name);
+
+  @override
+  String toString() => 'AnsiType.$_name';
+}
+
+/// Standard ANSI escape code for customizing terminal text output.
+///
+/// [Source](https://en.wikipedia.org/wiki/ANSI_escape_code#Colors)
+class AnsiCode {
+  /// The numeric value associated with this code.
```

Comment:
Mainly we're `toString`ing more often that we'd need to if we kept this as a String, and by having it as a number we're expressing that it can be used as a number - which it shouldn't be since it's an identifier.